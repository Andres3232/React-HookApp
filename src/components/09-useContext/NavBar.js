import React from 'react'
import {Link, NavLink } from 'react-router-dom'

export const NavBar = () => {
    return (
    <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
        <div className="container-fluid">
            <NavLink className={`nav-link ${ (navData) => navData.isActive && 'active' }`} to='/'>UseContext</NavLink> 
                <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                    <div className="navbar-nav">
                        <NavLink className={`nav-link ${ (navData) => navData.isActive && 'active' }`} to='/home'>Home</NavLink> 
                        <NavLink className={`nav-link ${ (navData) => navData.isActive && 'active' }`} to='/about'>Aboout</NavLink> 
                        <NavLink className={`nav-link ${ (navData) => navData.isActive && 'active' }`} to='/login'>Login</NavLink> 
                    </div>
                </div>
        </div>
     </nav>
    )
}
