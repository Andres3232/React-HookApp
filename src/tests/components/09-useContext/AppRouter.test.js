import { mount } from "enzyme";
import { AppRouter } from "../../../components/09-useContext/AppRouter";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe('pruebas en AppRouter', () => {

    const user = {
        id:1,
        name: 'Andres'
    }

    const wrapper = mount(
        <UserContext.Provider value={{
            user
        }}>
            <AppRouter/>

        </UserContext.Provider>
    );

  test('debe de mostrarse correctamente', () => {
    
    expect( wrapper ).toMatchSnapshot()

  });
  
});
