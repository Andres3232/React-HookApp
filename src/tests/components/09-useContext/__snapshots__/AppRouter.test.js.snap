// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en AppRouter debe de mostrarse correctamente 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        aria-current="page"
        className="nav-link navData => navData.isActive && 'active' active"
        href="/"
        onClick={[Function]}
      >
        UseContext
      </a>
      <div
        className="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            className="nav-link navData => navData.isActive && 'active'"
            href="/home"
            onClick={[Function]}
          >
            Home
          </a>
          <a
            className="nav-link navData => navData.isActive && 'active'"
            href="/about"
            onClick={[Function]}
          >
            Aboout
          </a>
          <a
            className="nav-link navData => navData.isActive && 'active'"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    <div>
      <h1>
        HomeScreen
      </h1>
      <hr />
      <pre>
        {
   "id": 1,
   "name": "Andres"
}
      </pre>
    </div>
  </div>
</div>
`;
